File: simple-java/src/main/java/com/jy/simple2/MyApplication.java
package com.jy.simple2;

import android.app.Application;

public class MyApplication extends Application {

    private static Application instance;

    public static Application getInstance() {
        return instance;
    }

    @Override
    public void onCreate() {
        super.onCreate();
        if (instance == null) {
            instance = this;
        }
    }
}

File: app/src/main/java/com/jy/simple/MyApplication.kt
package com.jy.simple

import android.app.Application


class MyApplication : Application() {

    companion object {
        private var instance: MyApplication? = null
        fun getInstance(): MyApplication = instance!!
    }


    override fun onCreate() {
        super.onCreate()
        if (instance == null) {
            instance = this
        }
    }
}